src/popup/components/Header.vue

```diff
@@ -187,6 +200,21 @@ export default defineComponent({
       ionRouter.navigate({ name: currentHomeRouteName.value }, 'back', 'push');
     }
 
+    watch(
+      () => props.show,
+      async (value) => {
+        if (value) {
+          // @ts-ignore
+          headerEl.value?.$el.classList.remove('hidden');
+        } else {
+          // wait for the page transition to finish
+          await new Promise((resolve) => setTimeout(resolve, 150));
+          // @ts-ignore
+          headerEl.value?.$el.classList.add('hidden');
+        }
+      },
+    );
```

Comment:
Manipulating classes with vanilla JS is a bad practice within Vue component. Maybe you coudl add another bool ref and change it's value based on the `show` property. And then change the `hidden` class within the template block. What do you think?